pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -201,16 +202,50 @@ func (is *InstanceService) InstanceCreate(name string, config *openstackconfigv1
 		}
 	}
 	userData := base64.StdEncoding.EncodeToString([]byte(cmd))
+	// if we are using Boot-from-volume, let's ignore image in base create option
+	image := config.Image
+	if config.RootVolume.Size > 0 {
+		image = ""
+	}
+
 	createOpts = servers.CreateOpts{
 		Name:             name,
-		ImageName:        config.Image,
+		ImageName:        image,
 		FlavorName:       config.Flavor,
 		AvailabilityZone: config.AvailabilityZone,
 		Networks:         nets,
 		UserData:         []byte(userData),
 		SecurityGroups:   config.SecurityGroups,
 		ServiceClient:    is.computeClient,
 	}
+
+	if config.RootVolume.Size > 0 {
+		var blocks []bootfromvolume.BlockDevice
+		block := bootfromvolume.BlockDevice{
+			SourceType:          bootfromvolume.SourceImage,
+			BootIndex:           0,
+			UUID:                config.Image,
```

Comment:
This only works if `config.Image` is specified as UUID, while we allow the Image to be the name.

Reply:
exactly.. this was found during my local test as well
I will figure out how to translate it ... or maybe a doc indication?